examples/archive.html

```diff
@@ -113,7 +113,31 @@ <h3 id="offcanvasRightLabel">Images</h3>
                 fetchedFrom.appendChild(fetchedFromUrl)
 
                 placeButton.classList.add('btn', 'btn-sm', 'btn-outline-secondary', 'place-button');
-                placeButton.innerHTML = 'Place on map';
+                placeButton.innerHTML = 'Place on map ';
+
+                document.addEventListener('click', (event) => {
+                    if(event.target.classList.contains('place-button')){
+                      const targetPlaceButton = event.target;
+                      
+                    if(!targetPlaceButton.getElementsByClassName('fa-spinner')[0]){
```

Comment:
Hi @RealRichi3 I see you're getting this to run only if it doesn't have a spinner, so it's run only once each time the `forEach` is run. But I see what @sainamercy is saying that this code was originally written to run just once, outside the `forEach`, and so anything outside this conditional will run many times. It just seems to me we might as well leave the `addEventListener("click")` where it originally was at the bottom, OR to change it so it's listening only for a click on the button itself, like: 

```js
placeButton.addEventListener('click', (event) => {
```

...instead of on the whole document (it used to be that way because we handled all clicks in a single listener instead of setting one up each time a button is created). Does that make sense? You could do either one but here we are kind of mixing the two. 

Otherwise this looks great. If you can try solving this and uploading a screen recording of it working, that would be super! Thanks both of you!

Reply:
Thanks @jywarren, so you mean i should change the event listener to target the button itself instead of the document